app/models/user.rb

```diff
@@ -12,4 +15,32 @@ class User < ApplicationRecord
   has_many :teams, through: :team_users
   has_many :comments
   has_many :watchers
+
+  def all_projects
+    company = Company.first
+    result = projects.ids
+    result2 = company.projects.joins(:teams).where(teams: { id: self.teams.pluck(:id) }).pluck(:id)
```

Comment:
Use descriptive variable names

Reply:
Instead of running 2 queries to retrieve IDs first and then 3rd to retrieve projects, directly retrieve projects on line 21 and 22 and return the merged result